lib/atomic.h

```diff
@@ -100,8 +100,8 @@ using std::atomic_fetch_and_explicit;
 using std::atomic_thread_fence;
 using std::atomic_signal_fence;
 
-#elif defined(HAVE_STDATOMIC_H) && !defined(__CC_ARM) && !defined(__arm__) && \
-      !defined(__STDC_NO_ATOMICS__)
+#elif defined(HAVE_STDATOMIC_H) && !defined(__STDC_NO_ATOMICS__) && \
+(!defined(__ARMCC_VERSION) || defined(__GNUC__))
```

Comment:
why do you need to add the || defined(__GNUC__) ?
I would expect that you only replace` !defined(__CC_ARM) && !defined(__arm__) `by  `!defined(__ARMCC_VERSION)`

Reply:
@arnopo It was considered that other compilers did not meet this condition, but now it seems that it is not necessary,  I have removed it.